src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,93 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.stream.IntStream;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkArrayCapacity();
+        elementData[size] = value;
```

Comment:
I see you only **check** Capacity here. But checking is not enough - array must grow if it is full.